harvester_e2e_tests/apis/test_support_bundle.py

```diff
@@ -104,6 +106,79 @@ def test_logfile_exists(self, support_bundle_state):
             f"Some file(s) not found, files: {matches}\npatterns: {patterns}"
         )
 
+    @pytest.mark.dependency(depends=["donwnload support bundle"])
+    def test_secret_file_exists(self, support_bundle_state):
+        ''' ref: https://github.com/harvester/tests/issues/603 '''
+        pattern = r"^.*/yamls/namespaced/fleet-local/v1/secrets\.yaml"
+        for fname in support_bundle_state.files:
+            if re.match(pattern, fname):
+                break
+        else:
+            raise AssertionError("secret file is not available in namespaced/fleet-local")
+
+        with ZipFile(support_bundle_state.fio, 'r') as zf:
+            ctx = zf.read(fname)
+
+        secret = yaml.safe_load(ctx)
+        assert secret.get('items')
+
+        fails = []
+        for it in secret['items']:
+            if it.get('type') != "rke.cattle.io/machine-plan" or not it.get('data'):
+                fails.append(it)
+
+        assert not fails, (
+            f"Got {len(fails)} incorrect item(s) from secret file,"
+            " expected type should be 'rke.cattle.io/machine-plan'"
+            " and `data` field should not be empty but got:\n"
+            f"{fails}"
+        )
+
+        support_bundle_state.fio.seek(0)
```

Comment:
Is this necessary?
If so, perhaps better move to `L#121`?

Reply:
It is necessary when following test cases going to read the `fio` again.  and yeah move to L121 would be good.